ci-operator/step-registry/fips-check/node-scan/fips-check-node-scan-commands.sh

```diff
@@ -0,0 +1,109 @@
+    if (( FRC == 0 )); then
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="0" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="${subteam}:Node scan of fips check should succeedded or skipped"/>
+</testsuite>
+EOF
+    else
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="1" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="${subteam}:Node scan of fips check should succeedded or skipped">
+    <failure message="">Fips node scan check failed</failure>
+  </testcase>
+</testsuite>
+EOF
+    fi
+}
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+set_proxy
+run_command "oc whoami"
+run_command "oc version -o yaml"
+
+# skip for ARM64
+mapfile -t node_archs< <(oc get nodes -o jsonpath --template '{range .items[*]}{.status.nodeInfo.architecture}{"\n"}{end}' | sort -R | uniq -d)
+echo "The architectures included in the current cluster nodes are ${node_archs[*]}"
+NON_SUPPORTED_ARCHES=(arm64)
+for arch in "${node_archs[@]}"; do
+    if [[ "${NON_SUPPORTED_ARCHES[*]}" =~ $arch ]]; then
+        echo "Skip this test since it doesn't support $arch."
+        exit 0
+    fi
+done
+
+# get a master node
+master_node_0=`oc get node -l node-role.kubernetes.io/master= --no-headers | grep -Ev "NotReady|SchedulingDisabled"| awk '{print $1}' | awk 'NR==1{print}'`
+if [[ -z $master_node_0 ]]; then
+    echo "Error master node0 name is null!\n."
+    exit 1
+fi
+# create a ns
+project="node-scan-$RANDOM"
+run_command "oc new-project $project --skip-config-write"
+if [ $? == 0 ]; then
+    echo "create $project project successfully" 
+else
+    echo "fail to create $project project."
+fi
+# check whether it is disconnected cluster
+oc label namespace "$project" security.openshift.io/scc.podSecurityLabelSync=false pod-security.kubernetes.io/enforce=privileged pod-security.kubernetes.io/audit=privileged pod-security.kubernetes.io/warn=privileged --overwrite=true
+cluster_http_proxy=`oc get proxy cluster -o=jsonpath='{.spec.httpProxy}'`
+cluster_https_proxy=`oc get proxy cluster -o=jsonpath='{.spec.httpProxy}'`
+attempt=0
+while true; do
+    out=$(oc --request-timeout=60s -n "$project" debug node/"$master_node_0" -- chroot /host bash -c "export http_proxy=$cluster_http_proxy; curl -sSI ifconfig.me --connect-timeout 5" 2> /dev/null || true)
+    if [[ $out == *"Via: 1.1"* ]]; then
+        echo "This is not a disconnected cluser"
+        break
+    fi
+    attempt=$(( attempt + 1 ))
+    if [[ $attempt -gt 3 ]]; then
```

Comment:
nit: you can use `>` inside `[[ ... ]]`

Reply:
Update to ">" will leading to lint check error. Details seen from https://www.shellcheck.net/wiki/SC2071. Thus changed it back.